src/js/components/DateInput/DateInput.js

```diff
@@ -97,6 +98,8 @@ const DateInput = forwardRef(
       schema ? valueToText(normalizedDate, schema) : undefined,
     );
 
+    const { icon: MaskedInputIcon, ...restOfInputProps } = inputProps || {};
+
```

Comment:
We should add a comment that changing the icon with inputProps is deprecated and also put a warning in the console noting this as well.
```suggestion
    // Setting the icon through `inputProps` is deprecated.
    // The `icon` prop should be used instead.
    const { icon: MaskedInputIcon, ...restOfInputProps } = inputProps || {};
    if (MaskedInputIcon) {
      console.warn(
        `Customizing the DateInput icon through inputProps is deprecated. 
Use the icon prop instead.`,
      );
    }

```